crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -597,6 +597,47 @@ impl<'w> EntityMut<'w> {
         unsafe { self.cell.get_mut() }
     }
 
+    /// Gets mutable access to the component of type `T` for the current entity.
+    /// Returns `None` if the entity does not have a component of type `T`.
+    ///
+    /// This only requires `&self`, and so may be used to get mutable access to multiple components.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_ecs::prelude::*;
+    /// #
+    /// #[derive(Component)]
+    /// struct X(usize);
+    /// #[derive(Component)]
+    /// struct Y(usize);
+    ///
+    /// # let mut world = World::default();
+    /// let mut entity = world.spawn((X(0), Y(0))).into_mutable();
```

Comment:
Should EntityWorldMut deref automatically into EntityMut?

Is `into_mutable` a new api? I haven't seen that before. I guess it's a new inner type shared between EntityWorldMut and FilteredEntityMut for generic mutable component access?